packages/kolibri-common/composables/useFacilities.js

```diff
@@ -0,0 +1,82 @@
+import { reactive, computed } from 'vue';
+import FacilityResource from 'kolibri-common/apiResources/FacilityResource';
+import useUser from 'kolibri/composables/useUser';
+import redirectBrowser from 'kolibri/utils/redirectBrowser';
+import FacilityDatasetResource from 'kolibri-common/apiResources/FacilityDatasetResource';
+import Lockr from 'lockr';
+
+const state = reactive({
```

Comment:
I would prefer that we not exactly map the vuex state here.

I think it would be cleaner to define each bit of state as a separate `ref`.

So:

```
const facilityConfig = ref({});
const facilities = ref([]);
const facilityId = ref(null);
```

Would then need to update the state access and update accordingly.

Reply:
If you want to keep them encapsulated and not externally editable, you could prefix their names with `_` and still return computed props instead.